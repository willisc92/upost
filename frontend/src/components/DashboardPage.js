import React from "react";
import { connect } from "react-redux";

class DashboardPage extends React.Component {
    render() {
        return (
            <div className="page-header">
                <div className="content-container">
                    {!this.props.isAuthenticated ? (
                        <h1 className="page-header__title">You must login.</h1>
                    ) : (
                        <h1 className="page-header__title">Welcome, ${localStorage.getItem("first_name")}!</h1>
                    )}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isAuthenticated: !!state.auth.token
    };
};

export default connect(mapStateToProps)(DashboardPage);
