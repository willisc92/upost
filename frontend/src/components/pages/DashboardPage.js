import React from "react";
import { connect } from "react-redux";
import { getAllInterests, startSetUserInterests } from "../../actions/interests";
import { startSetInterestRandomPosts } from "../../actions/posts";
import MyPostSummary from "../MyPostSummary";

export class DashboardPage extends React.Component {
    componentDidMount() {
        if (this.props.isAuthenticated) {
            this.props.startSetUserInterests().then(() => {
                this.props.startSetInterestRandomPosts(this.props.interests);
            });
        } else {
            this.props.getAllInterests().then(() => {
                this.props.startSetInterestRandomPosts(this.props.interests);
            });
        }
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        {!this.props.isAuthenticated ? (
                            <h1 className="page-header__title">You must login.</h1>
                        ) : (
                            <h1 className="page-header__title">{`Welcome, ${localStorage.getItem("first_name")}!`}</h1>
                        )}
                    </div>
                </div>
                <div className="content-container">
                    {this.props.interestRandomPosts.map((interestPosts) => {
                        return (
                            <div key={interestPosts.tag}>
                                <h1>{interestPosts.tag}</h1>
<<<<<<< HEAD
                                {/* <div className="list-parent"> */}
                                {interestPosts.posts.map((post) => {
                                    return (
                                        // <div key={post.post_id} className="list-box">
                                        <MyPostSummary post={post} pathName={`/post/${post.post_id}`} />
                                        // </div>
=======
                                {interestPosts.posts.map((post) => {
                                    return (
                                        <MyPostSummary
                                            key={post.post_id}
                                            post={post}
                                            pathName={`/post/${post.post_id}`}
                                        />
>>>>>>> react-pages
                                    );
                                })}
                            </div>
                            // </div>
                        );
                    })}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isAuthenticated: !!state.auth.token,
        interests: state.userInterests.userInterests,
        interestRandomPosts: state.posts.interestRandomPosts
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        getAllInterests: () => dispatch(getAllInterests()),
        startSetInterestRandomPosts: (interests) => dispatch(startSetInterestRandomPosts(interests)),
        startSetUserInterests: () => dispatch(startSetUserInterests())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DashboardPage);
