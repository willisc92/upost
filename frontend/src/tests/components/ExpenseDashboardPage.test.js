import React from "react";
import { shallow } from "enzyme";
import DashboardPage from "../../components/DashboardPage";

test("Should render ExpenseDashboardPage", () => {
    const wrapper = shallow(<DashboardPage />);
    expect(wrapper).toMatchSnapshot();
});
