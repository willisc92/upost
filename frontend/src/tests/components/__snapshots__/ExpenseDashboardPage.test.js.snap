// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseDashboardPage 1`] = `
<div>
  Dashboard Page Content
</div>
`;
